src/main/scala/chiseltest/exceptions.scala

```diff
@@ -17,3 +17,5 @@ class TimeoutException(message: String) extends ChiselTestException(message)
 
 // when interfacing with the testdriver before stepping the clock after moving to an earlier region
 class TemporalParadox(message: String) extends ChiselTestException(message)
+
+case class TestApplicationException(exitVal: Int, lastMessage: String) extends RuntimeException(lastMessage)
```

Comment:
Does it really make sense to have the last message here - or is it even guaranteed that the last message is useful? It might make more sense to just generically define this as a message (like the above exceptions), and whatever generates this exception should try to put something useful there?